### PR TITLE
Add Core Configuration and Project Structure for STM32

### PR DESCRIPTION
**Pull request by daviddaguero**
#### Summary:
This pull request introduces essential configuration files and code for the STM32 system, alongside the foundational project structure.

#### Key Changes:

1. **Configuration Code**:
   - Added `configuration.c` and `configuration.h` files.
   - Implemented setup functions for core components:
     - `system_clock_setup()`: Configures the system clock to 72 MHz using PLL and HSE.
     - `configure_gpio()`: Sets up GPIO pins for devices like the alarm, motor, manual and override switches, LED, fan, and various sensors.
     - `exti_setup()`: Configures EXTI interrupts for the override switch.
     - `config_i2c()`: Configures I2C1 for communication with external displays or peripherals.
     - `config_pwm()`: Sets up PWM signal generation for motor control.
     - `adc_setup()`: Initializes ADC for sensor input readings, including temperature and battery levels.

2. **Project Structure**:
   - Established the basic folder structure for the project (`src`, `include`, `lib`, etc.).
   - Added `platformio.ini` configuration file for PlatformIO integration.

This setup provides the groundwork for essential microcontroller operations and sensor interaction, setting up components for a robust and responsive system.

### CODE REVIEW

File: src/configuration.c [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    adc_disable_external_trigger_regular(ADC1);
+    adc_set_single_conversion_mode(ADC1);    // Single conversion per channel
+    adc_set_sample_time(ADC1, ADC_CHANNEL_TEMP_SENSOR, ADC_SMPR_SMP_55DOT5CYC); /*  // Sampling time
+
+    /* Calibrate ADC */
+    adc_power_on(ADC1);                      // Power on the ADC
+    adc_reset_calibration(ADC1);
+    adc_calibrate(ADC1);
+    
+}
```
Comment (sofiavalos) [inline src/configuration.c:153]: endline?

**Review by sofiavalos — CHANGES_REQUESTED** [review]

File: src/configuration.c (hunk 1 above)
Comment (daviddaguero) [reply src/configuration.c:153]: The endlines are just fine :-1:

**Review by sofiavalos — CHANGES_REQUESTED** [review]
delete build dir

**Review by sofiavalos — APPROVED** [review]
LGTM!🥳

**Review by lujanlaspina — APPROVED** [review]
🤞

**Pull request merged**